wisp-logging/src/main/kotlin/wisp/logging/SampledLogger.kt

```diff
@@ -0,0 +1,211 @@
+package wisp.logging
+
+import mu.KLogger
+import mu.Marker
+import wisp.sampling.Sampler
+
+internal class SampledLogger constructor(
+    override val underlyingLogger: KLogger,
+    private val sampler: Sampler
+) : KLogger by underlyingLogger {
```

Comment:
This is a change from the misk version which I believe had [accidentally introduced a bunch of infinitely recursive overrides](https://github.com/cashapp/misk/blob/master/misk-core/src/main/kotlin/misk/logging/SampledLogger.kt#L244-L262) because it wasn't delegating the the `underlyingLogger` in all cases.  By using [delegated properties](https://kotlinlang.org/docs/delegated-properties.html), we only need to override the what we want to be sampled and everything else just uses the underlying version by default.

The risk here is that the if a new function gets added to the `KLogger` interface, we won't know about it and it won't be sampled until we explicitly override it.  I don't think the likelihood of being bitten by that is high enough to worry, but happy to revert back to manual delegation if that is preferred.